plugins/modules/foreman_provisioning_template.py

```diff
@@ -326,8 +326,9 @@ def main():
         if module.state == 'present_with_defaults':
             module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
         if module.desired_absent:
-            if len(module_params.keys()) != 1:
-                module.fail_json(msg="When deleting all templates, there is no need to specify further parameters.")
+            further_params = set(module_params.keys()) - {'name', 'entity'}
```

Comment:
scope creep, but this check exists in many modules and should be in the helper

Reply:
The whole affects multiple should be.
That's my exercise to the reader these days.

Reply:
You mean the reader of https://github.com/theforeman/foreman-ansible-modules/issues/244?

Reply:
My new take on 'foreman_installation_medium' might help there.

Reply:
You mean "pull it out of the with api section so it can be moved to a helper"? yes, looks like it.